src/NServiceBus.Core/Pipeline/MutateInstanceMessage/MutateIncomingMessageBehavior.cs

```diff
@@ -15,7 +15,9 @@ class MutateIncomingMessageBehavior : Behavior<IIncomingLogicalMessageContext>
             var mutatorContext = new MutateIncomingMessageContext(current, context.Headers);
             foreach (var mutator in context.Builder.BuildAll<IMutateIncomingMessages>())
             {
-                await mutator.MutateIncoming(mutatorContext).ConfigureAwait(false);
+                await mutator.MutateIncoming(mutatorContext)
```

Comment:
My question was in another direction. Since we always eagerly create the message like in this example with 

```
($"{mutator.GetType().FullName}.MutateIncoming"
```

We do reflection and allocation of strings every time just to create a bit more helpful message. My vote is to remove the additional message. The memory and perf hit is not worth for such a small benefit.

Thoughts @Particular/nservicebus-maintainers